src/server/game/Chat/Chat.cpp

```diff
@@ -964,15 +1000,75 @@ ObjectGuid::LowType ChatHandler::extractLowGuidFromLink(char* text, HighGuid& gu
 std::string ChatHandler::extractPlayerNameFromLink(char* text)
 {
     // |color|Hplayer:name|h[name]|h|r
-    char* name_str = extractKeyFromLink(text, "Hplayer");
-    if (!name_str)
+    std::string name_str = extractKeyFromLink(text, "Hplayer");
```

Comment:
This here is still calling `char*     extractKeyFromLink(char* text, char const* linkType, char** something1 = nullptr);` which can return nullptr - assigning it to std::string will crash

Reply:
Yeah I shouldn't of touched that function, that was a mistake.